scripts/affected_targets

```diff
@@ -14,29 +14,63 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-# Finds all targets of a given kind affected by changes since origin/develop.
+# Finds all targets of a given kind affected by changes in a given commit range.
+
+SCRIPT_NAME=$(basename "$0")
 
 # type should be a bazel `kind` type. E.g. test or rule.
 # https://docs.bazel.build/versions/master/query.html#kind
 type="$1"
 
+# range should be a valid git diff/log range. E.g. origin/develop...HEAD
+range="$2"
+
+if [ -z "$type" ]; then
+  echo "Please provide a blaze rule type to query for."
+  exit 1
+fi
+
+if [ -z "$range" ]; then
+  echo "Please provide a diff range."
+  exit 1
+fi
+
+if [ $(git branch --list "$TARGET_BRANCH") ]; then
+  base_sha=$(git merge-base "$TARGET_BRANCH" HEAD)
+else
+  base_sha="HEAD^"
+fi
+
 # Generates a list of files that were modified since origin/develop.
 modified_files() {
-  git log --name-only --pretty=oneline --full-index $(git merge-base origin/develop HEAD)...HEAD \
+  git log --name-only --pretty=oneline --full-index "$range" \
     | grep -vE '^[0-9a-f]{40} ' \
     | sort \
     | uniq
 }
 
-# TODO: Add check for whether we need to rebuild everything.
+# Reads a list of files and generates a list of files that will affect the entire
+# build system.
+files_that_affect_everything() {
+  grep -e "\.bzl$" \
+    -e "WORKSPACE" \
+    -e "\.kokoro" \
+    -e "$SCRIPT_NAME" \
+    "$@"
+}
+
+if modified_files | files_that_affect_everything -q; then
+  echo "//..."
+  exit 0
+fi
 
 # Generates a list of targets that are affected by the files modified since origin/develop.
 directly_modified_targets() {
-  for line in $(modified_files); do
+  while read line; do
     label=$(bazel query $line 2>/dev/null)
```

Comment:
$line may need to be escaped here. I saw trouble with any headers that have '+' in the filename.  

This worked:  `label=$(bazel query \'$LINE\' 2>/dev/null)`

**Command-line testing:**

```bash
# SUCCESS =  single-quote for bash, double-quote for bazel
rsmoore-macbookpro1:material-components-ios rsmoore$ bazel query '"components/ActionSheet/src/ActionSheetThemer/MaterialActionSheet+ActionSheetThemer.h"'
INFO: Invocation ID: cd7fcccd-6bda-409c-b508-fdddd1b828b8
//components/ActionSheet:src/ActionSheetThemer/MaterialActionSheet+ActionSheetThemer.h

# FAILURE = double-quote for bash
rsmoore-macbookpro1:material-components-ios rsmoore$ bazel query "components/ActionSheet/src/ActionSheetThemer/MaterialActionSheet+ActionSheetThemer.h"
INFO: Invocation ID: 584b23ac-d676-460a-9c65-b08d7c5cf19b
ERROR: no such target '//:ActionSheetThemer.h': target 'ActionSheetThemer.h' not declared in package '' defined by /Users/rsmoore/git/material-components-ios/BUILD

# FAILURE = single-quote for bash
rsmoore-macbookpro1:material-components-ios rsmoore$ bazel query 'components/ActionSheet/src/ActionSheetThemer/MaterialActionSheet+ActionSheetThemer.h'
INFO: Invocation ID: 2bdd1ce4-376c-44be-8feb-74325be247b2
ERROR: no such target '//:ActionSheetThemer.h': target 'ActionSheetThemer.h' not declared in package '' defined by /Users/rsmoore/git/material-components-ios/BUILD
rsmoore-macbookpro1:material-components-ios rsmoore$
```

Reply:
Done.